test/viewmodels/comment_view_model_integration_test.dart

```diff
@@ -0,0 +1,149 @@
+import "package:collection/collection.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/comment/comment_data.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/ui/comment_post.dart";
+import "package:proxima/services/database/comment_repository_service.dart";
+import "package:proxima/services/database/firestore_service.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+import "package:proxima/viewmodels/comment_view_model.dart";
+
+import "../mocks/data/comment_data.dart";
+import "../mocks/data/firestore_post.dart";
+import "../mocks/data/firestore_user.dart";
+import "../mocks/data/geopoint.dart";
+
+void main() {
+  group("Testing comment view model", () {
+    late FakeFirebaseFirestore fakeFirestore;
+    late CommentDataGenerator commentDataGenerator;
+
+    late UserRepositoryService userRepository;
+    late CommentRepositoryService commentRepository;
+    late AutoDisposeFamilyAsyncNotifierProvider<CommentViewModel,
+        List<CommentPost>, PostIdFirestore> commentViewModelProvider;
+    late ProviderContainer container;
+
+    late PostIdFirestore postId;
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+      commentDataGenerator = CommentDataGenerator();
+
+      container = ProviderContainer(
+        overrides: [
+          firestoreProvider.overrideWithValue(fakeFirestore),
+        ],
+      );
+
+      userRepository = container.read(userRepositoryProvider);
+      commentRepository = container.read(commentRepositoryProvider);
+
+      final post = FirestorePostGenerator().generatePostAt(userPosition0);
+      await setPostFirestore(post, fakeFirestore);
+      postId = post.id;
+
+      commentViewModelProvider = commentListProvider(postId);
+    });
+
+    group("display of comments", () {
+      test("No comments are returned when the database is empty", () async {
+        final comments = await container.read(commentViewModelProvider.future);
+
+        expect(comments, isEmpty);
+      });
+
+      test("Multiple comments are returned when the database has comments",
+          () async {
+        const numberOwners = 3;
+        const numberComments = 19;
+
+        // Create and add the comments owner to the database
+        final owners =
+            FirestoreUserGenerator.generateUserFirestore(numberOwners);
+        for (final owner in owners) {
+          await userRepository.setUser(owner.uid, owner.data);
+        }
```

Comment:
I think that this can be extracted to a helper function, like the function `addPosts` of the `PostDataGenerator`.

Reply:
You are totally right.
Actually I found out that we already have a function `setUsersFirestore` that I can directly use here.
Thanks for pointing that out :+1: 

Fix in 08ea0b5b67a76bc271fda01bbe6a7c94e49f0249.

Reply:
This is not exactly what I meant, I thought of something closer to:
https://github.com/ProximaEPFL/proxima/blob/22e9a498134fbdcaec99dce2e5e1c95045eb3996/test/mocks/data/firestore_post.dart#L90C1-L99C1

However, I'm realising a function for two lines of code is a bit overkill. I'd therefore understand you stay like this, this is already much better! :)

Reply:
Oh ok I see, that's actually a good idea and could be useful in the future.
I implement this in 8ec59b4b04e97d79fb9142785fe8c0c8401fcafb.

Thanks for the suggestion :)

Reply:
Interestingly, in `PostDataGenerator`, `addPosts` is not static method: it is outside of the class. I think that `addComments` should not be inside the class out of consistency, but I'd understand if you want to keep this as is since this is, in my opinion, more coherent.
I moreover think it is missing a bit of documentation ;)

Reply:
I believe that `addPosts` is indeed located inside `FirestorePostGenerator`.
Thus I think that I will let `addUsers` inside `FirestoreUserGenerator` if that's ok for you.

I have also added a line of documentation :)
Done in 03b0684ec430e869d91b964108de1a1c51e83df9.

Reply:
Yes indeed, my bad! Thank you :)